tests/tools/test_convert_partition.py

```diff
@@ -0,0 +1,64 @@
+import os
+import tempfile
+
+import numpy as np
+import pytest
+
+from tools.distpartitioning import utils
+
+from tools.distpartitioning.convert_partition import _get_unique_invidx
+
+
+@pytest.mark.parametrize(
+    "num_nodes, num_edges, nid_begin, nid_end",
+    [
+        [4000, 40000, 0, 1000],
+        [4000, 40000, 1000, 2000],
+        [4000, 40000, 2000, 3000],
+        [4000, 40000, 3000, 4000],
+    ],
+)
+def test_get_unique_invidx(num_nodes, num_edges, nid_begin, nid_end):
+    # prepare data for the function
+    # generate synthetic edges
+    srcids = np.random.randint(0, num_nodes, (num_edges,))
+    dstids = np.random.randint(nid_begin, nid_end, (num_edges,))
+
+    # generate unique node-ids for any
+    # partition. This list should be sorted.
+    # This is equivilant to shuffle_nids in a partition
+    unique_nids = np.arange(nid_begin, nid_end)
+
+    # invoke the test target
+    uniques, idxes, src_ids, dst_ids = _get_unique_invidx(
```

Comment:
Can you compare the output with a implementation using np.unique?

Reply:
I don't understand. For every call do you want me to compare the workaround solution with numpy's unique ??